code/restorer.go

```diff
@@ -82,6 +85,30 @@ func (r Restorer) Restore() error {
 			continue
 		}
 		originFileName := filePath[:len(filePath)-len(failpointStashFileSuffix)]
+		rewritedContent, err := ioutil.ReadFile(originFileName)
+		if err != nil {
+			return err
+		}
+		originContent, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			return err
+		}
+		// Rewrite original file
+		rewriter := NewRewriter(filePath)
+		buffer := &bytes.Buffer{}
+		rewriter.SetOutput(buffer)
+		if err := rewriter.RewriteFile(filePath); err != nil {
+			return err
+		}
+
+		// Merge modifications after `failpoint-ctl enable`
+		patcher := diffmatchpatch.New()
+		diffs := patcher.DiffMain(buffer.String(), string(rewritedContent), true)
+		patches := patcher.PatchMake(diffs)
+		pathedContent, _ := patcher.PatchApply(patches, string(originContent))
```

Comment:
The Apply could error, no? Like if I deleted a `failpoint.Eval` invocation on the enabled file it will likely fail.

Reply:
`PatchApply` will not return errors, see: https://github.com/sergi/go-diff/blob/HEAD/diffmatchpatch/patch.go#L232:28

Reply:
Could we consider the apply failed to the returned array contained at least one `false` entry?

Reply:
addressed